app/helpers/competition_api.rb

```diff
@@ -23,6 +23,25 @@ def self.fetch_competition(competition_id)
     end
   end
 
+  def self.get_competition_info(competition_id)
+    competition_info = Rails.cache.fetch(competition_id, expires_in: 5.minutes) do
+      self.fetch_competition(competition_id)
+    end
+
+    if competition_info[:error] == false
+      competition_info[:competition_exists?] = true
```

Comment:
reconsider if we need these two booleans

Reply:
I think it makes sense to leave them in and then refactor the competition_api.rb once the PR is merged, as discussed.